[FIX/IMP] improve website content list view

xO-Tx opened the pull request:
After [1], The 'website_pages_list' view is used on website content records.

The goal of this PR is to:

- Tweak the website filter (added at [1]) to make it work for all website content
records (page,blog,...).

- Update the 'New Page' dialog to be able to select the website_id for the new
page.

[1]: https://github.com/odoo/odoo/pull/98937

task-2889981

robodoo commented:
[Pull request status dashboard](https://mergebot.odoo.com/odoo/odoo/pull/98941).

qsm-odoo reviewed (commented):
@xO-Tx I did not include your commit "set action to create forum posts" from this branch yet. I'll let you rebase this branch after the other work at https://github.com/odoo/odoo/pull/98937#pullrequestreview-1095806216 is merged :+1:

qsm-odoo commented:
@xO-Tx an update on this one? :)

qsm-odoo commented:
@xO-Tx Do you think this will be ready soon? :+1: (you can reply here on Github otherwise people may think I am speaking alone here :smile:)

xO-Tx commented:
> @xO-Tx Do you think this will be ready soon? +1 (you can reply here on Github otherwise people may think I am speaking alone here smile)

It's Ok for me now :+1:,  I was about to send you an update about it... here some remarks:

- I fixed some issues related to code sharing between list and kanban, added the views (just let me know if you need something to be changed on kanban views UI).

- I disabled the website filter on `appointment.type` since it's only a `website.published.mixin` (the use of `website_id` will trigger an error)

- The homepage icons on `website.page` views:

![Screenshot from 2022-09-16 18-25-55](https://user-images.githubusercontent.com/25032876/190685890-0e191b3d-fce7-4817-b893-d58ca1e00c3a.png)

Are invisible since [This update](https://github.com/odoo/odoo/commit/9c733d8c2214979ca77203942c139137caefebe8) which uses `homepage_url` to compute homepage (if url =  `/`, `is_homepage` field has a value == `False` but the page is used as homepage when `homepage_url` is not set in settings)

I kept the code for homepage icons in this PR since we can simply update the `_compute_is_homepage` method to have something like this:

```python
def _compute_is_homepage(self):
    website = self.env['website'].get_current_website()
    for page in self:
        page.is_homepage = page.url == (website.homepage_url or page.website_id == website and '/') 
```
@rdeodoo what do you think ?

qsm-odoo commented:
@xO-Tx With your branch:

- Go to pages manager
- Click on create
- Crash

... I am sure there are mostly good things in what you did but could you test a bit more and make sure everything is working? I'd like to merge this soon (I guess if we freeze I'll have to merge in master first and then backport because of the new views which is a bit annoying) but don't tell it's finished if things like that are still broken :grimacing: :wink:

xO-Tx commented:
> * Go to pages manager
> * Click on create
> * Crash

It was actually an issue with my last rebase (precisely after [the fetchwebsites() update](https://github.com/odoo/odoo/commit/7b16b0683895bf6d35cda0f27399df8ecc2e598a)) on "CREATE" dialog for `website.page`... fixed now :+1:

qsm-odoo reviewed (approved):
cc @xO-Tx 

- The modification of the blog post kanban views has been squashed into the commit introducing the JS code for the kanban views
- Moved the commit I can't merge in 16.0 directly (addition of kanban views) at https://github.com/odoo/odoo/pull/101174 (I'll merge that once this one forward-port and then I'll backport it)
- Rebased this on 16.0 and retargeted the PR + various commit message fixes
- Added a commit of my own in this one to add multi-edit everywhere. @xO-Tx I was not able to do it for the "jobs" because of a bug (see commit message), maybe you could try and fix it?
- I found another bug in this PR which I did not fix: I cannot use the website dropdown in the searchview for the "Appointments" records, it crashes, could you also have a look at that @xO-Tx ?

Thanks for the work :+1: 

@robodoo rebase-ff r+

robodoo commented:
Merge method set to rebase and fast-forward.

xO-Tx commented:
> * Added a commit of my own in this one to add multi-edit everywhere. @xO-Tx I was not able to do it for the "jobs" because of a bug (see commit message), maybe you could try and fix it?
> * I found another bug in this PR which I did not fix: I cannot use the website dropdown in the searchview for the "Appointments" records, it crashes, could you also have a look at that @xO-Tx ?

OK thanks, I'll check that :+1:

qsm-odoo commented:
@robodoo r+
